web/client/plugins/Widgets.jsx

```diff
@@ -40,37 +40,31 @@ const WidgetsView = connect(
 
 class Widgets extends React.Component {
      static propTypes = {
-         id: PropTypes.string,
-         enabled: PropTypes.bool,
-         limitDockHeight: PropTypes.bool,
-         fluid: PropTypes.bool,
-         zIndex: PropTypes.number,
-         dockSize: PropTypes.number,
-         position: PropTypes.string,
-         dimMode: PropTypes.string,
-         src: PropTypes.string,
-         style: PropTypes.object
+         enabled: PropTypes.bool
      };
      static defaultProps = {
-         id: "widgets-plugin",
-         enabled: true,
-         dockSize: 600,
-         limitDockHeight: true,
-         zIndex: 10000,
-         fluid: false,
-         dimMode: "none",
-         position: "right"
+         enabled: true
      };
     render() {
-        return (<ContainerDimensions>{({width, height}) => <WidgetsView width={width} height={height}/>}</ContainerDimensions> );
+        return this.props.enabled ? (<ContainerDimensions>{({width, height}) => <WidgetsView width={width} height={height}/>}</ContainerDimensions> ) : null;
 
     }
 }
 
-module.exports = {
-    WidgetsPlugin: assign(Widgets, {
+const WidgetsPlugin = connect(
+    createSelector(
+        state => checkConditionsSelector(state, [
```

Comment:
move this in maplayout selectors and name it better. rightPanelOpenSelector, bottomPanelOpenSelector and put them together here.